### PR TITLE
fix improper input value issue #646

### PR DESCRIPTION
**Pull request by umayangag**

### CODE REVIEW

**Review by dinukadesilva — APPROVED** [review]

File: results-tabulation-api/api/TallySheetVersionApi/__init__.py [hunk 1]
```diff
@@ -165,6 +166,14 @@ def get_by_id(tallySheetId, tallySheetVersionId):
 def create(tallySheetId, body):
     request_body = RequestBody(body)
     tally_sheet = TallySheet.get_by_id(tallySheetId=tallySheetId)
+
+    # validate user inputs to prevent XSS attacks
+    if not input_is_valid(request_body.get("content")):
+        raise InvalidInputException(
```
Comment (dinukadesilva) [inline results-tabulation-api/api/TallySheetVersionApi/__init__.py:172]: validating the special character and length should be applied only on "strValue" if it's not defined.

File: results-tabulation-api/exception/__init__.py [hunk 2]
```diff
@@ -23,3 +23,7 @@ def InternalServerErrorException(message="", code=None):
 
 def NotImplementedException(message="", code=None):
     raise ProblemException(501, "Not Implemented", message, "NotImplemented", code)
+
+
+def InvalidInputException(message="", code=None):
+    raise ProblemException(400, "Invalid Input", message, "Forbidden", code)
```
Comment (dinukadesilva) [inline results-tabulation-api/exception/__init__.py:29]: You could use one of the above instead adding new

File: results-tabulation-api/util/__init__.py [hunk 3]
```diff
@@ -156,3 +156,14 @@ def get_sum_of_all_and_nan_otherwise(array):
             result += val
 
     return result
+
+
+def input_is_valid(content_array):
```
Comment (dinukadesilva) [inline results-tabulation-api/util/__init__.py:161]: Better to rename to be more specific

File: results-tabulation-api/api/TallySheetVersionApi/__init__.py (hunk 1 above)
Comment (umayangag) [reply results-tabulation-api/api/TallySheetVersionApi/__init__.py:172]: all values are converted to str for validation. is it much better to leave it at that?

**Pull request merged**